### ISSUE TITLE
size adjustement required [nvidiaprime@pdcurtis]

### ISSUE THREAD

**Issue by MuhmdRaouf** [linuxmint/cinnamon-spices-applets#1552]
Cinnamon version: 3.6.6
Distribution - Mint 18.3
Graphics hardware *and* driver used: Nvidia-387

I'm using the Nvidia prime applet, but in the latest distro, it looks kinda weird maybe that's due the new dpi that used in cinnamon it gets more like squared shape.
![screenshot from 2017-12-08 13-10-02](https://user-images.githubusercontent.com/10425180/33763776-73f78642-dc1a-11e7-8a3c-caef3e0da6eb.png)

**Comment by NikoKrause** [comment]
@pdcurtis

**Comment by pdcurtis** [comment]
I think this is an effect of the icon having a two colour background and a similar background colour in the panel which makes it look odd in your screenshot. It looks fine on my machine with a different background. Try changing theme and see if that 'improves' it. I think you will find it is the same with the nvidia-prime applet.

**Comment by NikoKrause** [comment]
@pdcurtis Are you on Cinnamon 3.6.6?

**Comment by pdcurtis** [comment]
cinnamon --version
Cinnamon 3.4.6

**Comment by NikoKrause** [comment]
OK. The issue described here is a new issue in Cinnamon 3.6.6

**Comment by MuhmdRaouf** [comment]
The Wallpaper doesn't fix anything tho mate, the applet used to look like the green one too
That's the Nvidia apple that shows up after installing the driver.
![screenshot from 2017-12-12 16-13-26](https://user-images.githubusercontent.com/10425180/33889088-c1a39364-df57-11e7-8a1f-7977ab7bfee8.png)
that's how it looks using the applet.
![33763776-73f78642-dc1a-11e7-8a3c-caef3e0da6eb](https://user-images.githubusercontent.com/10425180/33889091-c55b0262-df57-11e7-9d5d-c11dbc49abdb.png)

**Comment by pdcurtis** [comment]
I do not understand this. Neither the icon or code has not been changed by me and it is all in two lines of code
            this.nvidea_icon = metadata.path + "/icons/nvidia.png"
and 
                this.set_applet_icon_path(this.nvidea_icon);

Has the Intel icon changed?

@NikoKrause Has there been a change in this.set_applet_icon_path() in 3.6.6 that can account for the loss of colour? 

Note: The Icon has always been square to save space

**Comment by NikoKrause** [comment]
?

I thought this issue is about the icon being squashed/square.
I know the icon with the default nvidia applet wasn't square in Cinnamon 3.4, but it's square in Cinnamon 3.6.

Why should change this the color? Isn't your applet supposed to use a gray icon?

**Comment by MuhmdRaouf** [comment]
Okie sorry that's my bad i didn't clearfiy things better, 
The green icon is the one comes with the driver(just icon not applet), the grey is the applet that located in mint repo, the grey icon used look as same as green one in cinnamon 3.4 and earlier, it just looks weird now when it gets squerd looking in cinnamon 3.6, i thought its something need some adjustments due the new dpi used in the new release of cinnamon, and that's why i report it, but its not bug after all if the dev intented  to make it like the in the first place to save space ^_^

**Comment by pdcurtis** [comment]
@NikoKrause My icon has always had the green colour and been square. I have just looked on Github and confirmed it has not changed and displays green and black on my machine under 3.4. Like you, I do not see why it should be any different but it obviously is from these screen-shots - it is very odd.

I want plenty of time when I change to Mint 18.3 as I need the machine for a while and it would be a problem if it fell over at present. Unfortunately one can't load Bumblebee on a LiveUSB.

@MuhmdRaouf Thanks for the clarification. Perhaps close the original Issue but I want to get to the bottom of why it now shows in gray-scale in any case.

If you find the icon shape a problem there is no reason why you should not replace it in cinnamon-spices-applets/bumblebee@pdcurtis/files/bumblebee@pdcurtis/icons with a suitable replacement called  nvidia.png after renaming the existing one to .bak

**Comment by NikoKrause** [comment]
The squashed icon in nvidia-prime@cinnamon.org applet is a known bug. I do have a fix for it, but need to test it. I don't have a Nvidia prime graphics card. If you can test it, this would be great.
I'll upload it here later.

**Comment by pdcurtis** [comment]
@MuhmdRaouf I have now upgraded to Mint 18.3 and Cinnamon 3.6.6 and my icon has not changed at all - it is still square and coloured. The other reason for the shape is for use on vertical panels - I keep one mainly for test purposes with my two applets designed to work on vertical panels. 
![image](https://user-images.githubusercontent.com/2843925/33931680-9cdd4c96-dfe8-11e7-9024-6e333ac41410.png)
@NikoKrause There are problems running nvidia-prime and bumblebee together. I tried to use nvidia-prime on a LiveUSB with remarkably little success so I will see what I can do to test your modification when you upload it but no promises. I can always try your new icon (if you do not mind) in bumblebee@pdcurtis

**Comment by NikoKrause** [comment]
Here's my patched nvidia applet version. Can you please check it out, also @MuhmdRaouf:

[nvidia-prime@cinnamon.org.zip](https://github.com/linuxmint/cinnamon-spices-applets/files/1556330/nvidia-prime.cinnamon.org.zip)

**Comment by MuhmdRaouf** [comment]
@NikoKrauseI have tried your version and still looks squished same as the one already exist, I just don't understand why the applet changed in the last release of cinnamon, and to be honest tho i only wanted the grey applet as it looks better in the taskbar, but i don't mind the green applet as the squished version looks kinda weird.

@pdcurtis I am talking about the nvidia-prime "applet" that comes the with distro and it always looks grey, the colored one shows up only after you install the nivida driver "nivdia icon not applet", and its look good as the pic showed earlier

**Comment by NikoKrause** [comment]
Have you restarted your machine after installing my patched version?

**Comment by NikoKrause** [comment]
Sorry,
there was a typo in the code. I hope this one solves the issue:
[nvidia-prime@cinnamon.org.zip](https://github.com/linuxmint/cinnamon-spices-applets/files/1558758/nvidia-prime.cinnamon.org.zip)

**Comment by MuhmdRaouf** [comment]
it looks so much better now, but kinda huge :rofl: 
Still much better than the squished square one 
![screenshot from 2017-12-14 14-33-21](https://user-images.githubusercontent.com/10425180/33993219-492c5046-e0de-11e7-8c35-eeb3a73e67ec.png)

**Comment by NikoKrause** [comment]
OK. I'll make it a little bit smaller. Does everything else works as expected?
Changing between Nvida and Intel, etc.?
If yes, I'll create a Pull Request to fix this.

**Comment by MuhmdRaouf** [comment]
Everything looks good and functional now, but as you know kinda too bigger 
![screenshot from 2017-12-14 15-45-11](https://user-images.githubusercontent.com/10425180/33995413-99c2c794-e0e6-11e7-9b4d-455c0b0e620a.png)

hope if u could make smaller size or at list tell me how to manipulate the code to change the icon size 
to make it near or little bit bigger of  the original icon size 
![screenshot from 2017-12-14 15-45-27](https://user-images.githubusercontent.com/10425180/33995506-e1c184c2-e0e6-11e7-85bc-fe2676b742c8.png)

**Comment by NikoKrause** [comment]
Hey, try to add the `-8` like I did in this line https://github.com/linuxmint/nvidia-prime-applet/pull/3/files#diff-0be14f5fed6e58cbe585e6dedd8b23e9R83

I think that's a pretty good size.

**Comment by MuhmdRaouf** [comment]
Thanks a lot mate :heart: 
and it is a good size indeed, its just sometimes u want to adjust things the way you like ^_^ 
thanks for helping me to manipulate the code.
Should i close the issue now though.

**Comment by NikoKrause** [comment]
I created a Pull Request. It will hopefully get merged soon.

**Comment by pdcurtis** [comment]
I am slightly confused. Was this issue actually about nvidiaprime@pdcurtis or the original nvidia-prime@cinnamon.org applet which has now been ably fixed?

As far as my two applets bumblebee@pdcurtis and nvidiaprime@pdcurtis for supporting an Optimus set up - the main purpose was to show a GPU temperature when the Nvidia GPU is active but I also strongly believe that the icon should be obvious at a glance (ie coloured and filling an entire square area) when the power hungry Nvidia GPU is active. 

I think the issue should be closed as nvidia@pdcurtis still works as designed under 3.6.6 and I do not plan to change.

@NikoKrause I still have no way to check your applet fully but it looks as if that is all sorted out and checked out fully by @MuhmdRaouf

**Comment by MuhmdRaouf** [comment]
Looks like the issue  was about nvidia-prime@cinnamon.org but i didn't notice until now,  am sorry i meant to make u confused, and mentioed ur work by mistake, but after all looks like the issue has been solved, thanks you mates @pdcurtis @NikoKrause  you have been really great help :heart: 
![screenshot from 2017-12-15 11-45-25](https://user-images.githubusercontent.com/10425180/34036332-88a7db0a-e18d-11e7-9261-70833511b162.png)
